File: src/extras-ksyntaxhighlighting/libwildcardmatcher.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_KSyntaxHighlighting__WildcardMatcher
#include <QString>
#include <wildcardmatcher.h>
#include "libwildcardmatcher.h"
#include "libwildcardmatcher.hxx"

bool KSyntaxHighlighting__WildcardMatcher_ExactMatch(libqt_string candidate, libqt_string wildcard) {
    QString candidate_QString = QString::fromUtf8(candidate.data, candidate.len);
    QString wildcard_QString = QString::fromUtf8(wildcard.data, wildcard.len);
    return KSyntaxHighlighting::WildcardMatcher::exactMatch(candidate_QString, wildcard_QString);
}

File: src/extras-ksyntaxhighlighting/libwildcardmatcher.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// ### [Upstream resources](https://api.kde.org/ksyntaxhighlighting-wildcardmatcher.html)
pub const KSyntaxHighlighting__WildcardMatcher = extern struct {
    /// ### [Upstream resources](https://api.kde.org/ksyntaxhighlighting-wildcardmatcher.html#exactMatch)
    ///
    /// ## Parameter(s):
    ///
    /// ` candidate: []const u8 `
    ///
    /// ` wildcard: []const u8 `
    ///
    pub fn ExactMatch(candidate: []const u8, wildcard: []const u8) bool {
        const candidate_str = qtc.libqt_string{
            .len = candidate.len,
            .data = candidate.ptr,
        };
        const wildcard_str = qtc.libqt_string{
            .len = wildcard.len,
            .data = wildcard.ptr,
        };
        return qtc.KSyntaxHighlighting__WildcardMatcher_ExactMatch(candidate_str, wildcard_str);
    }
};
